### PR TITLE
[close me] comments

### PR DESCRIPTION
**Pull request by paulfreund94**

### CODE REVIEW

File: rescript.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
 7. All new components should be Jsx3 components. This is done to be able to use the latest language features.
 8. When deciding between `thing->function->function` and `function(thing)->function` read it out and pick the one that makes more sense in a natural language, i.e. `Firebase.firestore(firebase)->Firebase.root("main")` and `maybeSomething->Belt.Option.getWithDefault("definitelySomething")`
 9. Liberally simplify functions from other modules at the top of your code. Eg prefer `a->mul(b)` to `a->Ethers.BigNumber.mul(b)` by declaring it `let {mul} = module(Ethers.BigNumber)`. Re-name them to avoid collisions.
-10. When defining functions keep the types definition in the function, not in the variable, to allow more freedom for type inference, i.e. `let a = (b: string, c: int): string => b === "ok"` over `let a: ((string, int) => boolean) = (b, c) => b === "ok"`
+10. When defining functions keep the types definition in the function, not in the variable, to allow more freedom for type inference, i.e. `let a = (b: string, c: int): string => b === "ok"` over `let a: ((string, int) => boolean) = (b, c) => b === "ok"` [What's the difference between these two for the compiler?]
```
Comment (paulfreund94) [inline rescript.md:12]: also, should the first one be `let a = (b: string, c: int): boolean => b === "ok"` as opposed to `let a = (b: string, c: int): string => b === "ok"`?

File: UI.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 1. Avoid put more than 5 React components in a single file. Use separate files for big components or components that are used in multiple places.
 2. Avoid files with only a few lines of code that are overly simple, rather try group them meaningfully.
-3. Put multiline styles in the top of the file.
+3. Put multiline styles in the top of the file. [not sure i agree with this, inline is easier unless you're repeating the style]
```
Comment (JasoonS) [inline UI.md:5]: Hmm, yeah this was for styled components more than tailwind, can delete this line.

File: UI.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
...
 4. Prefer template strings over string concatenation.
 5. Use SWR if fetching data (outside of the graph) that is useful to persist between pages.
-6. Use react context for state data between pages.
+6. Use react context for state data between pages. [Would appreciate some more info as to when to choose between SWR and react context / providers.]
```
Comment (JasoonS) [inline UI.md:8]: Ok, "state data" is things like, is the user logged in or which network they are on or things lilke that.

For other things that are just fetching data from somewhere, I think SWR is better. It is easier to think about, and you don't have to have any complexity related to contexts etc. So I think currently we are over-using the context.

Will write more about this.

File: rescript.md (hunk 1 above)
Comment (JasoonS) [reply rescript.md:12]: Yes, the first one is definitely a boolean, sorry about that.

I'm not sure super sure about how the compiler treats this one differently, it is copied from the template that I used as a reference. Maybe it is worth removing, I don't think it is actually a real concern.

File: rescript.md [hunk 4]
```diff
@@ -34,4 +34,4 @@ let rec len = (myList: list<'a>) =>
 
 ## Other considerations:
 
-1. Prefer using uncurried functions over curried (i. e. _`reduceU` over `reduce`_ in Belt.List*)*, it will help the compiler to produce cleaner error messages in cases when there is a parameters mismatch, which happens quite often during refactoring. Another benefit is that it works faster.
+1. Prefer using uncurried functions over curried (i. e. _`reduceU` over `reduce`_ in Belt.List*)*, it will help the compiler to produce cleaner error messages in cases when there is a parameters mismatch, which happens quite often during refactoring. Another benefit is that it works faster. [Nice didn't know this]
```
Comment (JasoonS) [inline rescript.md:37]: Yeah, grabbed it from the reference style-guide I was looking at. Wasn't something I thought about before.

It is just much nicer writing curried functions.

**Comment by JasoonS** [comment]
Thanks - updated

**Pull request closed without merging**